simpleclient_pushgateway/src/main/java/io/prometheus/client/exporter/PushGateway.java

```diff
@@ -54,15 +56,42 @@
  */
 public class PushGateway {
 
-  private final String address;
+  // Visible for testing
+  final String gatewayBaseURL;
   private static final int MILLISECONDS_PER_SECOND = 1000;
   /**
    * Construct a Pushgateway, with the given address.
    * <p>
    * @param address  host:port or ip:port of the Pushgateway.
    */
   public PushGateway(String address) {
-    this.address = address;
+    this(createURLSneakily("http://" + address));
+  }
+
+  /**
+   * Construct a Pushgateway, with the given URL.
+   * <p>
+   * @param serverBaseURL the base URL and optional context path of the Pushgateway server.
+   */
+  public PushGateway(URL serverBaseURL) {
+    this.gatewayBaseURL = URI.create(serverBaseURL.toString() + "/metrics/job/")
+      .normalize()
+      .toString();
+  }
+
+  /**
+   * Creates a URL instance from a String representation of a URL without throwing a checked exception.
+   * Required because you can't wrap a call to another constructor in a try statement.
+   *
+   * @param urlString the String representation of the URL.
+   * @return The URL instance.
+   */
+  private static URL createURLSneakily(final String urlString) {
+    try {
+      return new URL(urlString);
+    } catch (MalformedURLException e) {
+      throw new RuntimeException(e);
+    }
```

Comment:
This is my hack to avoid breaking the existing API, which would have happened otherwise due to introducing a checked exception.